detectors/node/opentelemetry-resource-detector-container/src/detectors/ContainerDetector.ts

```diff
@@ -94,22 +81,70 @@ export class ContainerDetector implements Detector {
       .trim()
       .split('\n')
       .find(s => s.includes(this.HOSTNAME));
-    const containerIdStr = str
-      ?.split('/')
-      .find(s => s.length === this.CONTAINER_ID_LENGTH);
-    return containerIdStr || '';
+
+    const strArray = str?.split('/') ?? [];
+    for (let i = 0; i < strArray?.length; i++) {
+      if (
+        strArray[i] === this.MARKING_PREFIX &&
+        strArray[i + 1]?.length === this.CONTAINER_ID_LENGTH
```

Comment:
for the last iteration, `i+1` is always out of range and will be undefined.
We can change the loop so it ends at `strArray.length - 1` instead of accessing index out of range.